### ISSUE TITLE
Monadic `blockRender`?

### ISSUE THREAD

**Issue by mrkgnao** [mmark-md/mmark#46]
I'd like to render LaTeX diagrams into images that are then inserted into the output. I have a function of the following form:

```haskell
-- | Render some LaTeX content to an image whose filename is the 
-- hash of the content. Return the path to the image.
renderToFile :: Text -> IO FilePath
renderToFile latex = do
   ...
   pure savePath
```

I think the interface I would like is to interpret code-blocks with some particular infostring (e.g. `render-latex`) as LaTeX to be rendered into a PNG. If `blockRenderM` existed, I could just do

```haskell
renderLatexExt :: Extension
renderLatexExt = blockRenderM $ \case
  CodeBlock (Just "render-latex") content 
    -> (\uri -> Image undefined uri Nothing) <$> renderToFile content
  x -> pure x
```

or something.

**Comment by mrkkrp** [comment]
I have exposed enough of the internal machinery in #116, so it should be possible to write arbitrary custom renders.

**Comment by emekoi** [comment]
I actually implemented #116 in a local fork of `mmark` to try to write a monadic renderer and I don't think it's sufficient since extensions and renderers are currently pure (they are basically records of `Endo`s) and specialized to produce `HtmlT Identity ()` so you can't change the underlying monad. Supporting monadic extensions and renderers amounts to replacing `Endo a` with `Kleisli m a a` so you get some type like `MMarkT m` and `MMark = MMarkT Identity`.

**Comment by mrkkrp** [comment]
@emekoi Could you open a PR for that?
